account_invoice_supplier_ref_unique/models/account_invoice.py

```diff
@@ -41,3 +43,31 @@ def _check_unique_supplier_invoice_number_insensitive(self):
     def _onchange_supplier_invoice_number(self):
         if not self.reference:
             self.reference = self.supplier_invoice_number
+
+    @api.model
+    def _prepare_refund(self, invoice, date_invoice=None,
+                        date=None, description=None, journal_id=None):
+        '''
+        The unique vendor invoice number cannot be passed to the credit note
+        in vendor bills
+        '''
+        vals = super(AccountInvoice, self)._prepare_refund(invoice,
+                                                           date_invoice,
+                                                           date,
+                                                           description,
+                                                           journal_id)
+
+        if invoice and invoice.type in ['in_invoice', 'in_refund'] and\
+                'supplier_invoice_number' in vals:
+            vals['supplier_invoice_number'] = ''
+
+        return vals
+
+    @api.multi
+    def copy(self, default=None):
+        '''
+        The unique vendor invoice number is not copied in vendor bills
+        '''
+        if self.type in ['in_invoice', 'in_refund']:
+            default = dict(default or {}, supplier_invoice_number='')
```

Comment:
Ok, @ThomasBinsfeld please leave copy=False on supplier_invoice_number and prevent copying the `reference` field for supplier invoices and refunds.